### ISSUE TITLE
project sequences across scaffolds when zoomed, contiguous + projection

### ISSUE THREAD

**Issue by nathandunn** [GMOD/Apollo#680]
- [x] view sequence when ref sequence (unable to do so now at all)
- [x] view sequence when single bookmark
- [x] view sequence when projected single bookmarked 
- [x] view sequence when "contiguous" bookmark
- [x] view sequence when projected "contiguous" bookmarked (both)
  - [x] **create unit test for multisequenceprojection::projectSequence()**
     - [x] create simple unit test for full overlapping services
     - [x] create simple unit test for substrings of middle, first (only), last (only)
- [x] handle projection over multiple small sequences (test)
  - [x] fix integration tests

**Comment by nathandunn** [comment]
```
41f-ed959e598899"}],"track":{"projection":"None","referenceTrack":"Official Gene Set v3.2","label":"Group1.1::GroupUn87","sequenceList":[{"name":"Group1.1"},{"name":"GroupUn87"}],"padding":0}}
| Error 2015-11-24 09:38:11,224 [clientInboundChannel-1] ERROR springwebsocket.GrailsSimpAnnotationMethodMessageHandler  - Unhandled exception
Message: /opt/apollo/honeybee/data/seq/7cf/4ac/72/GroupUn87-11.txt (No such file or directory)
    Line | Method
->>  146 | <init>                                        in java.io.FileInputStream
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|    260 | $tt__loadResidueForSequence                   in org.bbop.apollo.SequenceService
|    235 | $tt__getSequenceChunkForChunk . . . . . . . . in     ''
|    221 | $tt__getRawResiduesFromSequence               in     ''
|     59 | $tt__getResidueFromFeatureLocation . . . . .  in     ''
|     48 | $tt__getResiduesFromFeature                   in     ''
|    512 | $tt__getResiduesFromTranscript . . . . . . .  in org.bbop.apollo.TranscriptService
|   2027 | $tt__getResiduesWithAlterations               in org.bbop.apollo.FeatureService
|    880 | $tt__getResiduesWithAlterationsAndFrameshifts in     ''
|    995 | _do_setLongestORF                             in     ''
|    477 | $tt__setLongestORF . . . . . . . . . . . . .  in     ''
|    459 | _do_calculateCDS                              in     ''
|    436 | $tt__calculateCDS . . . . . . . . . . . . . . in     ''
|   1840 | $tt__deleteFeature                            in org.bbop.apollo.RequestHandlingService
```

**Comment by nathandunn** [comment]
@cmdcolin   had suggested using the JBrowse sequence types where you can directly include indexed FASTA.    I think that this drastically simplifies the code and thus also has the nice side-effect that it makes it far easier to integrate with the project libraries.

**Comment by nathandunn** [comment]
. . . sorry . should not have closed it.

**Comment by nathandunn** [comment]
I think that this will actually work as-is . . . except for the contiguous projections without using the FAI.  For that you have to calculate the chunk locations I believe.  To do that you need the proper offset anyway.  I have to have this working for other stuff.

**Comment by nathandunn** [comment]
This might be blocked until I get #51  done.   The sequence chunk stuff should just work . . . its mis-calculating the chunks right now.  

Basically it needs to know how to do this type of calculation:

```
http://localhost:8080/apollo/jbrowse/data/seq/50e/b7a/08/%7B%22padding%22:0,%20%22projection%22:%22None%22,%20%22referenceTrack%22:%22Official%20Gene%20Set%20v3.2%22,%20%22sequenceList%22:[%7B%22name%22:%22Group1.1%22%7D],%20%22label%22:%22Group1.1%22%7D:-1..-1-4.txt
```

that should be pretty easy as we're doing it for every other track right now.

**Comment by nathandunn** [comment]
I need to take a look as there are multiple side-effects to this right now

**Comment by nathandunn** [comment]
The original error is that "contiguous" projections (exon, or none) was using the last contiguous projection coordinate, instead of the full scaffold/sequence length.

**Comment by nathandunn** [comment]
Working . . attaching relevant screen shots.

**Comment by nathandunn** [comment]
Screenshots from both sides, projected and unprojected.   There is still an off-by-one error in the projected version (a minor fix in #673 ).  Still need to fix mismatched features (will create issue). 

![screen shot 2015-12-30 at 4 24 33 pm](https://cloud.githubusercontent.com/assets/751274/12059524/fc819198-af11-11e5-8726-086497242530.png)
![screen shot 2015-12-30 at 4 23 54 pm](https://cloud.githubusercontent.com/assets/751274/12059525/fc864940-af11-11e5-874d-c3e233f55c4d.png)
![screen shot 2015-12-30 at 4 23 49 pm](https://cloud.githubusercontent.com/assets/751274/12059529/fc8be094-af11-11e5-81ac-2f95bded2bff.png)
![screen shot 2015-12-30 at 4 22 54 pm](https://cloud.githubusercontent.com/assets/751274/12059526/fc88c670-af11-11e5-8a59-644672289e4d.png)
![screen shot 2015-12-30 at 4 22 13 pm](https://cloud.githubusercontent.com/assets/751274/12059528/fc8bbe70-af11-11e5-8765-70a4e152cc0d.png)
![screen shot 2015-12-30 at 4 21 44 pm](https://cloud.githubusercontent.com/assets/751274/12059527/fc89ad60-af11-11e5-8f0d-8c698d1983df.png)
![screen shot 2015-12-30 at 4 21 40 pm](https://cloud.githubusercontent.com/assets/751274/12059530/fc96dc9c-af11-11e5-9dc1-321c3aaade02.png)
![screen shot 2015-12-30 at 4 21 27 pm](https://cloud.githubusercontent.com/assets/751274/12059531/fc9cf924-af11-11e5-967f-0da219c32cfe.png)
![screen shot 2015-12-30 at 4 19 46 pm](https://cloud.githubusercontent.com/assets/751274/12059532/fc9d09be-af11-11e5-8856-a1707f028814.png)
